### ISSUE TITLE
Add p2ms as a recognized standard script type.

### ISSUE THREAD

**Issue by wardlem** [JBaczuk/bscript-parser#49]

**Comment by wardlem** [comment]
This PR requires a consideration of what a templating ASM syntax should look like.  Currently, it looks like `OP_HASH160 <20-byte-script-hash> OP_EQUAL`.  The template processor sees that the script begins with `20-byte` and enforces that the push_data value is 20 bytes.  This works ok, though it isn't perfect.  The templating language currently doesn't support opcode parameters, which are required in a p2ms template (for *m* and *n*). For this, I was considering `<OP_1-OP_16>` and it would see that it expects an op within that range.  The public keys also need to be repeated, and this needs a syntax.  I was considering, for the public keys `<33-byte-public-key...>` where the ... indicates repetition.

I am not sold on any of these choices.

**Comment by JBaczuk** [comment]
Can you explain how this would be used/exposed? Or maybe what you mean by templating syntax. I'm trying to figure out when I would want to actually type `<20-byte-script-hash>` instead of the actual hash.

**Comment by wardlem** [comment]
> Can you explain how this would be used/exposed? Or maybe what you mean by templating syntax. I'm trying to figure out when I would want to type <20-byte-script-hash> instead of the actual hash.

Yes.  Script types are recognized using special "template" scripts that have placeholder values where variable data is used in standard scripts.  For example, the template: `OP_HASH160 <20-byte-script-hash> OP_EQUAL` will match the script `OP_HASH160 52973f3519d5d248004767efb874aa2a3b2b37ce OP_EQUAL` which allows the script to be detected as a `p2sh pubKey` script.  You can see how this is implemented in the [templates.js](https://github.com/JBaczuk/bscript-parser/blob/master/src/templates.js) file.  This functionality is not currently directly exposed to the end user as a top-level export, though it could be in the future.

**Comment by JBaczuk** [comment]
Ok I get what your talking about. So for the `p2ms` why not just use a template like:

`<p2ms>`

Then the template processor knows to look for an `OP_M` which must be 0-16, then it looks for the last byte to be `OP_N` from 0-16, and `N` pubkeys.

Another alternative (which might be more descriptive if this template was ever exposed to the user):

`<OP_M> <pubkey> ... <OP_N>`

Though this seems to make more work for the template processor.

**Comment by wardlem** [comment]
> Another alternative (which might be more descriptive if this template was ever exposed to the user):
>
> <OP_M> <pubkey> ... <OP_N>
>
> Though this seems to make more work for the template processor.

It is a little more work for the template processor, but I believe that this is a natural way to describe what a p2ms looks like.  I vote for this.
